spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -949,12 +952,16 @@ else if (valueAttribute.indexOf('.') != -1) {
 					+ "disabled or header name is not explicitly provided via @Header annotation.");
 			String headerRetrievalExpression = "headers['" + headerName + "']";
 			String fullHeaderExpression = headerRetrievalExpression + relativeExpression;
-			String fallbackExpression = (annotationAttributes.getBoolean("required")
-					&& !methodParameter.getParameterType().getName().equals("java.util.Optional"))
-					? "T(org.springframework.util.Assert).isTrue(false, 'required header not available: "
-					+ headerName + "')"
-					: "null";
-			return headerRetrievalExpression + " != null ? " + fullHeaderExpression + " : " + fallbackExpression;
+			if (annotationAttributes.getBoolean("required")
+					&& !methodParameter.getParameterType().equals(Optional.class)) {
```

Comment:
Seems for me we can't backport this to `4.3.x`.
I see your comment in the PR description, but it doesn't include this change